pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/AbstractNode.java

```diff
@@ -425,22 +425,91 @@ public void remove() {
         // Detach current node of its parent, if any
         final Node parent = jjtGetParent();
         if (parent != null) {
-            parent.removeChildAtIndex(jjtGetChildIndex());
-            jjtSetParent(null);
+            parent.remove(jjtGetChildIndex());
         }
+    }
+
+    @Override
+    public void remove(final int index) {
```

Comment:
the semantics of this is weird... `remove()` removes a node from it's parent, but `remove(int)` removes a child from self?

Reply:
Yes. I think it would be nice to just call `node.remove()`. 
However, as the other rewrite methods (i.e. `insert` & `replace`) are called from the parent node to perform an operation over a child node at a given index, I would choose to stay with `remove(int)` instead of with `remove()`.

Reply:
The problem is, [`node.remove()` already exists](https://github.com/pmd/pmd/blob/master/pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/Node.java#L273-L276). Actually, you added it yourselves. So we end up with:

* `node.remove()` removes self from parent
* `node.remove(int)` removes child from self

This lends itself way too much for confusion. Tidy up the API and be consistent. We either always work on the children of a node, or on the node itself, but all other methods should be no more.

I'm not particularly happy however at removing a method we just added in 6.0.0. I fear some lack of proper planning on how the feature will work, by focusing too early on the abstract operations on the node (bottom-up), before actually figuring out what the higher layers of abstraction would require (top-down).

We can survive such deprecation if it's the right thing to do for PMD, but I'd really like to be certain we won't keep deprecating newly added methods as we move forward.